test/local_sensitivity/callbacks.jl

```diff
@@ -56,7 +58,24 @@ dstuff = ForwardDiff.gradient(
 @test dp2 ≈ dstuff[3:6]
 
 
-# callback with no effect except saving the state
+function dg!(out,u,p,t,i)
+  (out.=-1)
+end
+
+
+cb2 = DiffEqSensitivity.track_callbacks(CallbackSet(cb),prob.tspan[1])
+sol_track = solve(prob,Tsit5(),u0=u0,p=p,callback=cb2,tstops=[5.0],abstol=1e-14,reltol=1e-14,saveat=0.1)
+cb_adj = DiffEqSensitivity.setup_reverse_callbacks(cb2,BacksolveAdjoint())
+
+adj_prob = ODEAdjointProblem(sol_track,BacksolveAdjoint(),dg!,sol_track.t,nothing,
+						 callback = cb2,
+						 abstol=1e-14,reltol=1e-14)
+adj_sol = solve(adj_prob, Tsit5(), abstol=1e-14,reltol=1e-14)
+@test adj_sol(sol1.t)[end-1:end,:] ≈ sol1[:,:]
+@test du01 ≈ -adj_sol[1:2,end]
+@test dp1 ≈ adj_sol[3:6,end]
+
+### callback with no effect except saving the state
```

Comment:
Making this one work is also a bit tricky.. Probably because of the non-uniqueness of the checkpoints?

Reply:
oh interesting. Maybe that's it. I didn't look too deep into why it failed.

Reply:
This one fails because  `t` is not unique in
https://github.com/SciML/DiffEqSensitivity.jl/blob/beb0366d1992758f131cc7c95210fd20371f672a/src/adjoint_common.jl#L277

Therefore, the callbacks in the adjoint solution (for `dg`) stop to be triggered after the reverse callback is applied. (See the difference in the two plots. First one is "no effect except saving the state"; second one is "no effect")
![with_save](https://user-images.githubusercontent.com/42201748/103001228-d76c2b00-452c-11eb-9cc2-96d48de04f63.png)
![without_save](https://user-images.githubusercontent.com/42201748/103001238-db984880-452c-11eb-80f2-371e63b94baf.png)


So a possible fix could be to use
```_t = unique(t)
cur_time = Ref(length(_t))```
and to add another callback for the duplicates.